### PR TITLE
Update Longhorn Backing Image IP Plan for Storage Network

### PR DESCRIPTION
**Pull request by WebberHuang1118**
Related issue:
https://github.com/harvester/harvester/issues/4244

### CODE REVIEW

File: docs/advanced/storagenetwork.md [hunk 1]
```diff
@@ -189,8 +194,40 @@ status:
 
 #### Step 2
 
-- Check if all Longhorn `instance-manager-e` and `instance-manager-r` are ready and networks are correct.
-- Check if Annotations `k8s.v1.cni.cncf.io/network-status` has an interface named `lhnet1` and the IP address is in the IP range.
+Check if all Longhorn `instance-manager-e`, `instance-manager-r` and `backing-image-manager` pods are ready and networks are correct.
```
Comment (guangbochen) [inline docs/advanced/storagenetwork.md:197]:
```suggestion
Verify the readiness of all Longhorn `instance-manager-e`, `instance-manager-r`, and `backing-image-manager` pods, and confirm that their networks are correctly configured.

Execute the following command to inspect a pod's details:

```

File: docs/advanced/storagenetwork.md [hunk 2]
```diff
@@ -189,8 +194,40 @@ status:
...
+
+```bash
+kubectl -n longhorn-system describe pod <pod-name>
+```
+
+If it appears similar event as following, the Storage Network should run out of IPs.
```
Comment (guangbochen) [inline docs/advanced/storagenetwork.md:203]:
```suggestion
If you encounter an event resembling the following, it indicates that the Storage Network might have run out its available IPs:
```

File: docs/advanced/storagenetwork.md [hunk 3]
```diff
@@ -189,8 +194,40 @@ status:
...
+
+:::note
+
+The same error could be encountered during image uploading/downloading, if Storage Network is out of IPs. Please delete related images, and reconfigure Storage Network with sufficient IP range.
+
+:::
+
+#### Step 3
+
+Check if Annotations `k8s.v1.cni.cncf.io/network-status` has an interface named `lhnet1` and the IP address is in the IP range.
```
Comment (guangbochen) [inline docs/advanced/storagenetwork.md:230]:
```suggestion
Check the `k8s.v1.cni.cncf.io/network-status` annotations and ensure that an interface named `lhnet1` exists, with an IP address within the designated IP range.
```

File: versioned_docs/version-v1.1/advanced/storagenetwork.md [hunk 4]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
 	- Please refer Networking page for more details and configure `Cluster Network` and `VLAN Config` but not `Networks`.
 - IP range for Storage Network
 	- IP range should not conflict or overlap with Kubernetes cluster networks(`10.42.0.0/16`, `10.43.0.0/16`, `10.52.0.0/16` and `10.53.0.0/16` are reserved).
-	- IP range should be IPv4 CIDR format and 4 times the number of your cluster nodes. Longhorn will use 2 IPs per node, and the upgrade process will run two versions of Longhorn simultaneously. It will consume 4 IPs per node in the upgrade process.
-	- If your cluster has 250 nodes, the IP range should be larger than `/22`.
+	- IP range should be IPv4 CIDR format and Longhorn pods use Storage Network as following:
```
Comment (guangbochen) [inline versioned_docs/version-v1.1/advanced/storagenetwork.md:56]: please refer to v1.2 changes for the v1.1 doc, thanks.

File: docs/advanced/storagenetwork.md [hunk 5]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
-	- If your cluster has 250 nodes, the IP range should be larger than `/22`.
+	- IP range should be IPv4 CIDR format and Longhorn pods use Storage Network as following:
+    - `instance-manger-e` and `instance-manager-r` pods: 2 IPs per node, and the upgrade will require two versions of it (the old version ones will be tear down after upgrade success).
+    - `backing-image-ds` pods: for processing every on-the-fly uploading/downloading backing image data soure. These pods would be tear down as the images complete downloading or uploading.
+    - `backing-image-manager` pods: 1 IP per disk, and the upgrade will require two versions of it (the old version ones will be tear down after upgrade success).
+    - The required number of IPs could be calulated from a simple formula: `Required Number of IPs = Number of Nodes * 4 + Number of Disks * 2 + Number of Uploading/Downloading images`
+	- For example, if your cluster has 5 nodes, each node has two disks and ten images are going to be uploaded simultaneously. The IP range should be larger than `/26` (`5 * 4 + 5 * 2 * 2 + 10 = 50`).
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:61]:
```suggestion
	- For example, if your cluster has 5 nodes, each node has 2 disks and 10 images will be uploaded simultaneously, the IP range should be larger than `/26` (5 x 4 + 5 x 2 x 2 + 10 = 50).
```

File: docs/advanced/storagenetwork.md [hunk 6]
```diff
@@ -146,7 +151,7 @@ After applying Harvester's Storage Network setting, Harvester will stop all pods
 
 Harvester will also create a new NetworkAttachmentDefinition and update the Longhorn Storage Network setting.
 
-Once Longhorn setting is updated, Longhorn will restart all `instance-manager-r` and `instance-manager-e` to apply the new network configuration, and Harvester will restart pods.
+Once Longhorn setting is updated, Longhorn will restart all `instance-manager-r`, `instance-manager-e`  and `backing-image-manager` pods to apply the new network configuration, and Harvester will restart pods.
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:154]:
```suggestion
Once the Longhorn setting is updated, Longhorn will restart all `instance-manager-r`, `instance-manager-e`, and `backing-image-manager` pods to apply the new network configuration, and Harvester will restart the pods.
```

File: docs/advanced/storagenetwork.md (hunk 1 above)
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:197]:
```suggestion
Check if all Longhorn `instance-manager-e`, `instance-manager-r`, and `backing-image-manager` pods are ready and networks are correct.
```

File: docs/advanced/storagenetwork.md (hunk 2 above)
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:203]:
```suggestion
If the output is similar to the following one, the Storage Network has run out of IPs.
```

File: docs/advanced/storagenetwork.md [hunk 7]
```diff
@@ -189,8 +194,40 @@ status:
...
+  Warning  FailedCreatePodSandBox  2m58s                  kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for
+ sandbox "04e9bc160c4f1da612e2bb52dadc86702817ac557e641a3b07b7c4a340c9fc48": plugin type="multus" name="multus-cni-network" failed (add): [longhorn-system/ba
+cking-image-ds-default-image-lxq7r/7d6995ee-60a6-4f67-b9ea-246a73a4df54:storagenetwork-sdfg8]: error adding container to network "storagenetwork-sdfg8": erro
+r at storage engine: Could not allocate IP in range: ip: 172.16.0.1 / - 172.16.0.6 / range: net.IPNet{IP:net.IP{0xac, 0x10, 0x0, 0x0}, Mask:net.IPMask{0xff,
+0xff, 0xff, 0xf8}}
+
+  ....
+```
+
+Please reconfigure Storage Network with sufficient IP range.
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:220]:
```suggestion
Please reconfigure the Storage Network with a sufficient IP range.
```

File: docs/advanced/storagenetwork.md [hunk 8]
```diff
@@ -189,8 +194,40 @@ status:
...
+
+:::note
+
+The same error could be encountered during image uploading/downloading, if Storage Network is out of IPs. Please delete related images, and reconfigure Storage Network with sufficient IP range.
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:224]:
```suggestion
If the Storage Network has run out of IPs, you might encounter the same error during image uploads/downloads. In this case, delete the related images and reconfigure the Storage Network with a sufficient IP range.
```

File: docs/advanced/storagenetwork.md (hunk 3 above)
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:230]:
```suggestion
Check if the annotation `k8s.v1.cni.cncf.io/network-status` has an interface named `lhnet1` and the IP address is in the IP range.
```

**Review by vickyhella — CHANGES_REQUESTED** [review]
Some suggestions on the grammar.
By the way, I only commented on the v1.2 doc, so please also bring the changes to the v1.1 doc. ^^

Update: I didn't notice Guangbo's comments when I reviewed this PR. You can look into Guangbo's comments first and I will take a second look afterward.

File: docs/advanced/storagenetwork.md [hunk 9]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
 	- Please refer Networking page for more details and configure `Cluster Network` and `VLAN Config` but not `Networks`.
 - IP range for Storage Network
 	- IP range should not conflict or overlap with Kubernetes cluster networks(`10.42.0.0/16`, `10.43.0.0/16`, `10.52.0.0/16` and `10.53.0.0/16` are reserved).
-	- IP range should be IPv4 CIDR format and 4 times the number of your cluster nodes. Longhorn will use 2 IPs per node, and the upgrade process will run two versions of Longhorn simultaneously. It will consume 4 IPs per node in the upgrade process.
-	- If your cluster has 250 nodes, the IP range should be larger than `/22`.
+	- IP range should be IPv4 CIDR format and Longhorn pods use Storage Network as following:
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 1 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 2 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 3 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: versioned_docs/version-v1.1/advanced/storagenetwork.md (hunk 4 above)
Comment (WebberHuang1118) [reply versioned_docs/version-v1.1/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md [hunk 10]
```diff
@@ -10,7 +10,7 @@ title: "Storage Network"
 
 Harvester uses Longhorn as its built-in storage system to provide block device volumes for VMs and Pods. If the user wishes to isolate Longhorn replication traffic from the Kubernetes cluster network (i.e. the management network) or other cluster-wide workloads. Users can allocate a dedicated storage network for Longhorn replication traffic to get better network bandwidth and performance.
 
-For more informations, please refer [Longhorn Storage Network](https://longhorn.io/docs/1.3.2/advanced-resources/deploy/storage-network/)
+For more informations, please refer [Longhorn Storage Network](https://longhorn.io/docs/1.4.3/advanced-resources/deploy/storage-network/)
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308519157, thanks.

File: docs/advanced/storagenetwork.md [hunk 11]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
 	- Please refer Networking page for more details and configure `Cluster Network` and `VLAN Config` but not `Networks`.
 - IP range for Storage Network
 	- IP range should not conflict or overlap with Kubernetes cluster networks(`10.42.0.0/16`, `10.43.0.0/16`, `10.52.0.0/16` and `10.53.0.0/16` are reserved).
-	- IP range should be IPv4 CIDR format and 4 times the number of your cluster nodes. Longhorn will use 2 IPs per node, and the upgrade process will run two versions of Longhorn simultaneously. It will consume 4 IPs per node in the upgrade process.
-	- If your cluster has 250 nodes, the IP range should be larger than `/22`.
+	- IP range should be IPv4 CIDR format and Longhorn pods use Storage Network as following:
+    - `instance-manger-e` and `instance-manager-r` pods: 2 IPs per node, and the upgrade will require two versions of it (the old version ones will be tear down after upgrade success).
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308525772, thanks.

File: docs/advanced/storagenetwork.md [hunk 12]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - `backing-image-ds` pods: for processing every on-the-fly uploading/downloading backing image data soure. These pods would be tear down as the images complete downloading or uploading.
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308525772, thanks.

File: docs/advanced/storagenetwork.md [hunk 13]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - `backing-image-manager` pods: 1 IP per disk, and the upgrade will require two versions of it (the old version ones will be tear down after upgrade success).
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308525772, thanks.

File: docs/advanced/storagenetwork.md [hunk 14]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - The required number of IPs could be calulated from a simple formula: `Required Number of IPs = Number of Nodes * 4 + Number of Disks * 2 + Number of Uploading/Downloading images`
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308525772, thanks.

File: docs/advanced/storagenetwork.md (hunk 5 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308525772, thanks.

File: docs/advanced/storagenetwork.md (hunk 6 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 1 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308530167, thanks.

File: docs/advanced/storagenetwork.md (hunk 2 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308530799, thanks.

File: docs/advanced/storagenetwork.md (hunk 7 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 8 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md:227]: Updated per suggestion, thanks.

File: docs/advanced/storagenetwork.md (hunk 3 above)
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: duplicated as https://github.com/harvester/docs/pull/406#discussion_r1308538931, thanks.

File: docs/advanced/storagenetwork.md [hunk 15]
```diff
@@ -29,6 +29,7 @@ There are some prerequisites before configuring the Harvester Storage Network se
     - `kubectl get -A vmi`
 - All pods that are attached to Longhorn Volumes should be stopped.
     - Users could skip this step with the Harvester Storage Network setting. Harvester will stop Longhorn-related pods automatically.
+- All the on-going image uploadings or downloadings shoulde be completed or deleted.
```
[the remark this answers is not in the document]
Comment (WebberHuang1118) [reply docs/advanced/storagenetwork.md]: Updated per suggestion, thanks.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

File: docs/advanced/storagenetwork.md [hunk 16]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
 	- Please refer Networking page for more details and configure `Cluster Network` and `VLAN Config` but not `Networks`.
 - IP range for Storage Network
 	- IP range should not conflict or overlap with Kubernetes cluster networks(`10.42.0.0/16`, `10.43.0.0/16`, `10.52.0.0/16` and `10.53.0.0/16` are reserved).
-	- IP range should be IPv4 CIDR format and 4 times the number of your cluster nodes. Longhorn will use 2 IPs per node, and the upgrade process will run two versions of Longhorn simultaneously. It will consume 4 IPs per node in the upgrade process.
-	- If your cluster has 250 nodes, the IP range should be larger than `/22`.
+	- IP range should be IPv4 CIDR format and Longhorn pods use Storage Network as follows:
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:56]:
```suggestion
	- IP range should be in IPv4 CIDR format and Longhorn pods use Storage Network as follows:
```

File: docs/advanced/storagenetwork.md [hunk 17]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - `instance-manger-e` and `instance-manager-r` pods: These require 2 IPs per node. During an upgrade, two versions of these pods will exist (old and new), with the old version being deleted once the upgrade is successful.
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:57]:
```suggestion
    - `instance-manger-e` and `instance-manager-r` pods: These require 2 IPs per node. During an upgrade, two versions of these pods will exist (old and new), and the old version will be deleted once the upgrade is successful.
```

File: docs/advanced/storagenetwork.md [hunk 18]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - `backing-image-ds` pods: These are employed to process on-the-fly uploading and downloading of backing image data sources. These pods will be removed once image uploads or downloads are completed.
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:58]:
```suggestion
    - `backing-image-ds` pods: These are employed to process on-the-fly uploads and downloads of backing image data sources. These pods will be removed once image uploads or downloads are completed.
```

File: docs/advanced/storagenetwork.md [hunk 19]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - `backing-image-manager` pods: 1 IP per disk, similar to the instance manager pods, two versions of these will coexist during an upgrade, and the old ones will be removed after the successful upgrade
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:59]:
```suggestion
    - `backing-image-manager` pods: 1 IP per disk, similar to the instance manager pods. Two versions of these will coexist during an upgrade, and the old ones will be removed after the upgrade is completed.
```

File: docs/advanced/storagenetwork.md [hunk 20]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+    - The required number of IPs can be calculated using a simple formula: `Required Number of IPs = Number of Nodes * 4 + Number of Disks * 2 + Number of Uploading/Downloading images`
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:60]:
```suggestion
    - The required number of IPs is calculated using a simple formula: `Required Number of IPs = Number of Nodes * 4 + Number of Disks * 2 + Number of Images to Download/Upload`
```

File: docs/advanced/storagenetwork.md [hunk 21]
```diff
@@ -52,8 +53,12 @@ kubectl apply -f https://raw.githubusercontent.com/harvester/harvester/v1.1.0/de
...
+	- For example, if your cluster has five nodes, each node has two disks, and ten images will be uploaded simultaneously. The IP range should be greater or equal to `/26` (`5 * 4 + 5 * 2 * 2 + 10 = 50`).
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:61]:
```suggestion
	- For example, if your cluster has five nodes, each node has two disks, and ten images will be uploaded simultaneously, the IP range should be greater than or equal to `/26` (`5 * 4 + 5 * 2 * 2 + 10 = 50`).
```

File: docs/advanced/storagenetwork.md [hunk 22]
```diff
@@ -189,8 +194,43 @@ status:
...
+Verify the readiness of all Longhorn `instance-manager-e`, `instance-manager-r`, and `backing-image-manager` pods, and confirm that their networks are correctly configured.
+
+Execute the following command to inspect a pod's details:
+
+
+```bash
+kubectl -n longhorn-system describe pod <pod-name>
+```
+
+If you encounter an event resembling the following, it indicates that the Storage Network might have run out its available IPs:
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:206]:
```suggestion
If you encounter an event resembling the following one, the Storage Network might have run out of its available IPs:
```

File: docs/advanced/storagenetwork.md [hunk 23]
```diff
@@ -189,8 +194,43 @@ status:
...
+0xff, 0xff, 0xf8}}
+
+  ....
+```
+
+Please reconfigure the Storage Network with a sufficient IP range.
+
+:::note
+
+The same error could be encountered during image uploading/downloading, if Storage Network is out of IPs. Please delete related images, and reconfigure Storage Network with sufficient IP range.
```
Comment (vickyhella) [inline docs/advanced/storagenetwork.md:227]:
```suggestion
If the Storage Network has run out of IPs, you might encounter the same error when you upload/download images. Please delete the related images and reconfigure the Storage Network with a sufficient IP range.
```

**Review by vickyhella — COMMENTED** [review]
Added comments to the updated commit, and please also apply the suggestions to the v1.1 doc accordingly.

**Review by vickyhella — APPROVED** [review]
LGTM and thanks!

**Review by bk201 — APPROVED** [review]
LGTM, thanks!

**Review by LucasSaintarbor — APPROVED** [review]
LGTM 👍 Merging now

**Pull request merged**